### PR TITLE
feat(batch): add fargate Runtime Platform properties to ECS Fargate C…

### LINKED ISSUE #26484: (aws-batch-alpha): Construct Props RuntimePlatform
### Describe the feature

The property [RuntimePlatform](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-batch-jobdefinition-containerproperties.html#cfn-batch-jobdefinition-containerproperties-runtimeplatform) is not present in the [interface IEcsContainerDefinition](https://docs.aws.amazon.com/cdk/api/v2/docs/@aws-cdk_aws-batch-alpha.IEcsContainerDefinition.html)

### Use Case

I would want to use AWS Batch with Fargate on ARM64 as describe in this news : https://aws.amazon.com/about-aws/whats-new/2023/07/aws-batch-fargate-linux-arm64-windows-x86-containers-cli-sdk/

>To do that customers will need to create ARM64 or Windows job definition using RegisterJobDefinition API of AWS Batch. ARM64-based and Windows-based containers are supported in all Fargate on-demand compute environments.

### Proposed Solution

_No response_

### Other Information

_No response_

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.88.0

### Environment details (OS name and version, etc.)

All OS

### PR DESCRIPTION
**Pull request by xazhao**
The property [RuntimePlatform](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-batch-jobdefinition-containerproperties.html#cfn-batch-jobdefinition-containerproperties-runtimeplatform) is not present in the AWS Batch ECS Fargate Job Definition. This PR adds flatten properties fargateCpuArchitecture and fargateOperatingSystemFamily to the ECS Fargate Job Definition in AWS Batch.

Closes #26484.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk-testing/framework-integ/test/aws-batch/test/integ.ecs-job-definition.js.snapshot/tree.json [hunk 1]
```diff
@@ -1616,7 +1620,7 @@
               }
             },
             "constructInfo": {
-              "fqn": "@aws-cdk/aws-batch-alpha.EcsJobDefinition",
+              "fqn": "aws-cdk-lib.aws_batch.EcsJobDefinition",
```
Comment (jfuss) [inline packages/@aws-cdk-testing/framework-integ/test/aws-batch/test/integ.ecs-job-definition.js.snapshot/tree.json:1623]: Do we know why this changed?

File: packages/aws-cdk-lib/aws-batch/lib/ecs-container-definition.ts [hunk 2]
```diff
@@ -1009,6 +1023,20 @@ export interface EcsFargateContainerDefinitionProps extends EcsContainerDefiniti
    * @default - 20 GiB
    */
   readonly ephemeralStorageSize?: Size;
+
+  /**
+   * The vCPU architecture of Fargate Runtime.
+   *
+   * @default - X86_64
```
Comment (jfuss) [inline packages/aws-cdk-lib/aws-batch/lib/ecs-container-definition.ts:1030]: For my own understanding, why set defaults for this and `fargateOperatingSystemFamily`?

File: packages/@aws-cdk-testing/framework-integ/test/aws-batch/test/integ.ecs-job-definition.ts [hunk 3]
```diff
@@ -60,6 +60,8 @@ new batch.EcsJobDefinition(stack, 'ECSFargateJobDefn', {
     memory: Size.mebibytes(32768),
     ephemeralStorageSize: Size.gibibytes(100),
     fargatePlatformVersion: FargatePlatformVersion.LATEST,
+    fargateCpuArchitecture: ecs.CpuArchitecture.ARM64,
+    fargateOperatingSystemFamily: ecs.OperatingSystemFamily.LINUX,
```
Comment (jfuss) [inline packages/@aws-cdk-testing/framework-integ/test/aws-batch/test/integ.ecs-job-definition.ts:64]: Is it guaranteed that the `batch` `EcsFaragetContainerDefintion` will always match `ecs`? I am wondering if there is a case where ECS needs some update and then batch needs to follow. Meaning there would be a time where batch doesn't support all types ECS supports.

I might be looking at this in the wrong way or have an miss understanding on how things connect together within CDK.

**Review by jfuss — COMMENTED** [review]
All comments are mostly for my understanding of the change. Otherwise, seems like a straight forward change.

File: packages/@aws-cdk-testing/framework-integ/test/aws-batch/test/integ.ecs-job-definition.js.snapshot/tree.json (hunk 1 above)
Comment (xazhao) [reply packages/@aws-cdk-testing/framework-integ/test/aws-batch/test/integ.ecs-job-definition.js.snapshot/tree.json:1623]: The last change of this file was to stabilizing (convert alpha to regular module). So the snapshot would be before stabilizing which was still using `aws-batch-alpha`

File: packages/aws-cdk-lib/aws-batch/lib/ecs-container-definition.ts (hunk 2 above)
Comment (xazhao) [reply packages/aws-cdk-lib/aws-batch/lib/ecs-container-definition.ts:1030]: My understanding is the default value should be same as the default value in [CFN resource](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-batch-jobdefinition-runtimeplatform.html).

File: packages/@aws-cdk-testing/framework-integ/test/aws-batch/test/integ.ecs-job-definition.ts (hunk 3 above)
Comment (xazhao) [reply packages/@aws-cdk-testing/framework-integ/test/aws-batch/test/integ.ecs-job-definition.ts:64]: I saw it was already using enum values from ECS definition so I didn't really think about it... Looking at the current values they're the same but I'm not sure if that's guaranteed or it's documented somewhere.
If we can generate all these values from the schema in the future, then we can be confident about the values here.

File: packages/aws-cdk-lib/aws-batch/README.md [hunk 4]
```diff
@@ -479,6 +479,21 @@ jobDefn.container.addVolume(batch.EcsVolume.efs({
 }));
 ```
 
+### Running a ECS workflow with Fargate container
```
Comment (paulhcsun) [inline packages/aws-cdk-lib/aws-batch/README.md:482]:
```suggestion
### Running an ECS workflow with Fargate container
```

**Review by paulhcsun — APPROVED** [review]

**Pull request merged**